code/controllers/subsystem/SSshuttles.dm

```diff
@@ -303,4 +310,110 @@ SUBSYSTEM_DEF(shuttle)
 	var/turf/spawn_location = pick(supply_shuttle_turfs)
 	new /obj/structure/closet/crate/mail(spawn_location)
 
+// load an alternative shuttle in at the appropriate landmark.
+/datum/controller/subsystem/shuttle/proc/load_template(datum/map_template/shuttle/S)
+	// load shuttle template, centred at shuttle import landmark,
+	if(loading_shuttle_at_preview_template)
+		CRASH("A shuttle was already loading at the preview template when another was loaded")
+
+	loading_shuttle_at_preview_template = TRUE
+	var/turf/landmark_turf = get_turf(locate("landmark*Shuttle Import"))
+	S.load(landmark_turf, centered = TRUE)
+
+	var/affected = S.get_affected_turfs(landmark_turf, centered = TRUE)
+
+	var/mobile_docking_ports = 0
+	var/obj/docking_port/mobile/port
+	// Search the turfs for docking ports
+	// - We need to find the mobile docking port because that is the heart of
+	//   the shuttle.
+	// - We need to check that no additional ports have slipped in from the
+	//   template, because that causes unintended behaviour.
+	for(var/T in affected)
+		for(var/obj/docking_port/P in T)
+			if(istype(P, /obj/docking_port/mobile))
+				port = P
+				mobile_docking_ports++
+				if(mobile_docking_ports > 1)
+					qdel(P, force = TRUE)
+					log_world("Map warning: Shuttle Template [S.mappath] has multiple mobile docking ports.")
+				else if(!port.timid)
+					// The shuttle template we loaded isn't "timid" which means
+					// it's already registered with the shuttles subsystem.
+					// This is a bad thing.
+					WARNING("Template [S] is non-timid! Unloading.")
+					port.jumpToNullSpace()
+					loading_shuttle_at_preview_template = FALSE
+					return
+
+			if(istype(P, /obj/docking_port/stationary))
+				log_world("Map warning: Shuttle Template [S.mappath] has a stationary docking port.")
+
+	if(port)
+		loading_shuttle_at_preview_template = FALSE
+		return port
+
+	var/msg = "load_template(): Shuttle Template [S.mappath] has no mobile docking port. Aborting import."
+
+	for(var/T in affected)
+		var/turf/T0 = T
+		T0.contents = null
+
+	message_admins(msg)
+	WARNING(msg)
```

Comment:
We don't change anything about the message here, right? Seems a bit weird to be split up from the messages

Reply:
That's why we send it out to both with the same var, unless I'm misunderstanding you?